daemon/container.go

```diff
@@ -98,8 +98,6 @@ func (daemon *Daemon) Register(c *container.Container) error {
 
 	daemon.containers.Add(c.ID, c)
 	daemon.idIndex.Add(c.ID)
```

Comment:
I would say we've never checked errors here before, and the errors that could be produced are already checked earlier on before we send to truncindex.
The only really valid error is if something couldn't be inserted, but in the end is not actionable to the user, and does not affect the operation of the container in anyway, just lookup by prefix.

It probably wouldn't hurt, but it won't help either.